mlir/test/Dialect/LLVMIR/nvvm-test-range.mlir

```diff
@@ -0,0 +1,28 @@
+// RUN: mlir-opt -int-range-optimizations -canonicalize %s | FileCheck %s
```

Comment:
It's considered bad practice to run multiple passes in the same test. Is there no way to annotate the IR with these inferred ranges?

Reply:
NVM, this analysis has no test pass...
This is far from ideal, as this test now not only depends on canonicalizations and an optimization pass, but also on other dialects.

Reply:
I see `-int-range-optimizations -canonicalize` are used together in our tests, so I followed the same practice. 
https://github.com/llvm/llvm-project/blob/main/mlir/test/Dialect/Arith/int-range-interface.mlir#L1

But I agree with you having multiple pass in a test. I can remove to `-canonicalize` if it helps.

Reply:
I changed the test, removed `-canonicalize` and kept `-int-range-optimizations` only